### ISSUE TITLE
Add Delay to Binding properties

### ISSUE THREAD

**Issue by brianlagunas** [microsoft/microsoft-ui-xaml#4260]
# Proposal: Add Delay to Binding Properties

I propose the addition of a new property to both Binding and x:Bind named **Delay**

## Summary

The **Delay** property will get or set the amount of time, in milliseconds, to wait before updating the binding source after the value on the target changes.

## Rationale
This **Delay** property is available in WPF. This is a short coming that keeps the feature gap large enough to limit porting of applications or providing the ability to limit network chatter within applications.

Imagine a simple scenario of using a slider to asynchronously load or filter records. As you slide you must filter or load your dataset. You may use a control like a slider

```html
<Slider Value="{Binding NumberOfRecords, Delay=250}" Minimum="0" Maximum="10000" />
```
In this example, the Slider binding will not update the source of the binding until after 250 milliseconds of the value being set. This prevents unnecessary chatter and processes which keep the application running with the best possible performance.

Currently without this, the application is forced to make service calls loading large amounts of data for every single tick of the slider as it slides from 0 to n-number. Not a good experience.

**Comment by dotMorten** [comment]
I've often lacked this feature to make bindings - especially on sliders and textboxes that can cause some expensive layout updates to happen - more effective.

**Comment by StephenLPeters** [comment]
Agreed,  this would be a cool addition to bindings. You might be able to acheive something similar to this using Storyboards, the animations types have a BeginTime property that is a delay before they execute. https://docs.microsoft.com/en-us/dotnet/api/system.windows.media.animation.timeline.begintime?view=net-5.0

**Comment by brianlagunas** [comment]
@StephenLPeters this slider is just a simple example. It would be easy to implement the loading of data when the thumb is released by doing something like this:

```html
<Slider x:Name="slider" Value="{Binding NumberOfRecords, UpdateSourceTrigger=Explicit}"  />
```

Add an event handler for the thumb being released by the mouse/pointer
```cs
slider.AddHandler(Thumb.PointerReleasedEvent, new PointerEventHandler(ThumbReleased), true);
```

Explicitly update the binding to the source t invoke any actions such as loading data
```cs
        private void ThumbReleased(object sender, PointerRoutedEventArgs e)
        {
            var expression = _sliderNumOfRecords.GetBindingExpression(Slider.ValueProperty);
            expression.UpdateSource();
        }
```

However, the request for the Delay spans more than just this simple example.

**Comment by michael-hawker** [comment]
Yeah, we added a [`Debounce`](https://docs.microsoft.com/dotnet/api/microsoft.toolkit.uwp.ui.dispatcherqueuetimerextensions.debounce) method to the Toolkit to help with this type of scenario.

**Comment by SoggyBottomBoy** [comment]
I've got a similar scenario with a colorpicker in which the color is bound to a property that on change re-renders some graphics. Currently dragging the color causes many re-renders. I want it to appear real time so would rather not trigger an update on close or something similar. In the past I have achieved something similar using Delay.
